widget/button.go

```diff
@@ -661,6 +661,49 @@ func (b *Button) Click() {
 	}
 }
 
+// Press presses the button emulating a Mouse Left click
+func (b *Button) Press() {
+	b.init.Do()
+
+	offx := b.widget.Rect.Dx()
+	offy := b.widget.Rect.Dy()
+
+	// This means that there are some pixels that are not clickable
+	if b.mask != nil {
+		offx = offx / 2
+		offy = offy / 2
+	}
+	b.hovering = true
+	b.widget.MouseButtonPressedEvent.Fire(&WidgetMouseButtonPressedEventArgs{
+		Widget:  b.widget,
+		Button:  ebiten.MouseButtonLeft,
+		OffsetX: offx,
+		OffsetY: offy,
+	})
+}
+
+// Release releases the button emulating a Mouse Left release
+func (b *Button) Release() {
+	b.init.Do()
+
+	offx := b.widget.Rect.Dx()
+	offy := b.widget.Rect.Dy()
+
+	// This means that there are some pixels that are not clickable
+	if b.mask != nil {
+		offx = offx / 2
+		offy = offy / 2
+	}
+	b.hovering = false
+	b.widget.MouseButtonReleasedEvent.Fire(&WidgetMouseButtonReleasedEventArgs{
+		Widget:  b.widget,
+		Inside:  true,
+		Button:  ebiten.MouseButtonLeft,
+		OffsetX: offx,
+		OffsetY: offy,
+	})
+}
+
 func (b *Button) handleSubmit() {
```

Comment:
I think we should refactor this method too using your new Press and Release functions.

Something like this works even if the variable name sucks now and can get cleaned up further:
```
func (b *Button) handleSubmit() {
	if input.KeyPressed(ebiten.KeyEnter) || input.KeyPressed(ebiten.KeySpace) {
		if !b.justSubmitted && b.focused {
			b.justSubmitted = true
			b.Press()
		}
	} else if b.justSubmitted {
		b.Release()
		b.justSubmitted = false
	}
}
```